src/main.py

```diff
@@ -138,13 +138,41 @@ async def handle_event(event: Dict, event_context, project_id_owner: Optional[st
 
 async def process_project_metrics(context: Context, project_id: str, services: List[GCPService]):
     context.log(project_id, f"Starting processing...")
+    await check_x_goog_user_project_header_permissions(context, project_id)
     ingest_lines = await fetch_ingest_lines_task(context, project_id, services)
     fetch_data_time = time.time() - context.start_processing_timestamp
     context.fetch_gcp_data_execution_time[project_id] = fetch_data_time
     context.log(project_id, f"Finished fetching data in {fetch_data_time}")
     await push_ingest_lines(context, project_id, ingest_lines)
 
 
+async def check_x_goog_user_project_header_permissions(context: Context, project_id: str):
+    if project_id in context.use_x_goog_user_project_header:
+        return
+
+    service_usage_booking = os.environ['SERVICE_USAGE_BOOKING'] if 'SERVICE_USAGE_BOOKING' in os.environ.keys() \
+        else 'source'
+    if service_usage_booking != 'destination':
+        context.use_x_goog_user_project_header[project_id] = False
+        return

```

Comment:
Maybe I misunderstood, but I thought that we want to perform the check by default, and have an option to override it with env variable to avoid API calls. Now SERVICE_USAGE_BOOKING == "destination" is required to perform check at all.
@pawelsiwek what's you opinion?

Reply:
I applied fixes requested in your other comments. As I understood it, the following must be true to enable the per-project billing:
- `SERVICE_USAGE_BOOKING` env var is set to `destination`
- SA must have permission to set the `x-goog-user-project` header
Otherwise nothing changes in billing.
Happy to change this logic when clarified by @pawelsiwek